lib/api/LogManagerImpl.cpp

```diff
@@ -807,12 +807,53 @@ namespace MAT_NS_BEGIN
     void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            LOG_WARN("Attempting to set nullptr as DataInspector");
+            return;
+        }
+
+        auto itDataInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const std::shared_ptr<IDataInspector>& currentInspector)
+        {
+            return strcmp(typeid(*dataInspector).name(),typeid(*currentInspector).name()) == 0;
```

Comment:
Strongly discourage RTTI. Many consumers (including Office) disable this functionality intentionally as it bloats the binary. Further, it's not clear to me what you're trying to do here? Guarantee that no more than one instance of the same type of IDataInspector is registered? I can imagine multiple instances of the same type trying to send data to different endpoints, which should be supported.

Reply:
Consider uniqueifying by name, otherwise don't bother?

Reply:
changed to uniqueify by name